lib/hyperclient/link.rb

```diff
@@ -167,9 +167,17 @@ def _uri_template
 
     def http_method(method, body = nil)
       @resource = begin
-        response = Futuroscope::Future.new do
-          @entry_point.connection.run_request(method, _url, body, nil)
-        end
+        do_async = @entry_point.async_options[:enabled]
```

Comment:
I've updated this to do `if @entry_point.options[:async]` directly. Would you prefer I add an explicit `async` getter allowing this code to do `if @entry_point.async` instead?